src/work/PokerGame.sol

```diff
@@ -0,0 +1,226 @@
+// SPDX-License-Identifier: MIT
+pragma solidity ^0.8.17;
+
+// Importing necessary contracts and libraries from OpenZeppelin and LitProtocol
+import "@openzeppelin/contracts/token/ERC721/ERC721.sol";
+import "@openzeppelin/contracts/security/ReentrancyGuard.sol";
+import "@openzeppelin/contracts/utils/Counters.sol";
+import "@litprotocol/contracts/LitAccessControl.sol";
+
+// Main contract for managing a poker game
+contract PokerGame is ERC721, ReentrancyGuard, LitAccessControl {
+    using Counters for Counters.Counter; // Using Counters for managing unique game IDs
+
+    Counters.Counter private _gameIds; // Counter for game IDs
+
+    uint8 constant MAX_PLAYERS = 6; // Maximum number of players per game
+    uint256 constant INITIAL_CHIP_COUNT = 1000; // Initial chip count for each player
+
+    // Enum representing the possible states of the game
+    enum GameState { WAITING, ACTIVE, FINISHED }
+    // Enum representing the possible actions a player can take
+    enum PlayerAction { NONE, CALL, RAISE, FOLD }
+
+    // Struct representing a player in the game
+    struct Player {
+        address addr; // Player's address
+        uint256 chipCount; // Player's chip count
+        PlayerAction lastAction; // Player's last action
+        bool isActive; // Whether the player is active in the game
+    }
+
+    // Struct representing a poker game
+    struct Game {
```

Comment:
if you want to add a seed for deck randomness, you can add a private mapping to map each game to a seed. But this needs to be private to prevent people from accessing the seed

Reply:
Note that when this is initialised the seed can be seen in the txn call data